sdk/Lusid.Sdk/Utilities/LusidExceptionHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using Lusid.Sdk.Client;
+using RestSharp;
+
+namespace Lusid.Sdk.Utilities
+{
+    /// <summary>
+    /// Handles the generation of LUSID exceptions after receiving the ApiResponse
+    /// </summary>
+    public static class LusidExceptionHandler
+    {
+        /// <summary>
+        /// Generate exceptions from the ApiResponse when ResponseStatus is an Error,
+        /// and StatusCode has no content or is less than 400
+        /// </summary>
+        /// <param name="methodName">The name of the method</param>
+        /// <param name="response">The ApiResponse</param>
+        public static Exception CustomExceptionFactory(string methodName, IApiResponse response)
+        {
+            var status = (int)response.StatusCode;
+            
+            // Throw if status code is >= 400
+            if (status >= 400)
+            {
+                return new ApiException(status,
+                    $"Error calling {methodName}: {response.RawContent}",
+                    response.RawContent, response.Headers);
+            }
+
+            // Throw whenever an internal SDK error has been thrown. This will result in ResponseStatus = Error
+            if (response.ResponseStatus == ResponseStatus.Error)
+            {
+                return new ApiException(500,
+                    $"Internal SDK error occured when calling {methodName}: {response.ErrorText}",
+                    response.InternalException.StackTrace, response.Headers);
+
+                // TODO: Since we can't log this potentially sensitive information.
+                // consider sending the full response body here into a safe storage for debugging
+            }
+            return null;
```

Comment:
It is possible that LUSID responds with a `200` statusCode but we end up with a `500`, do you think this is a good result? Do you think it is misleading?

Reply:
Yeah I've thought about it and it's a fair point. Might be better to keep the status code the same as the response on the API. This is still an exception either way